include/zephyr/net/ethernet.h

```diff
@@ -1287,6 +1288,75 @@ static inline bool net_eth_is_vlan_interface(struct net_if *iface)
 #define ETH_NET_L3_REGISTER(name, ptype, handler) \
 	NET_L3_REGISTER(&NET_L2_GET_NAME(ETHERNET), name, ptype, handler)
 
+enum net_eth_mac_type {
+	NET_ETH_MAC_DEFAULT = 0,
+	NET_ETH_MAC_RANDOM,
+	NET_ETH_MAC_NVMEM,
+	NET_ETH_MAC_STATIC,
+};
+
+struct net_eth_mac_config {
+	enum net_eth_mac_type type;
+	uint8_t addr[NET_ETH_ADDR_LEN];
+	uint8_t addr_len;
+	struct nvmem_cell cell;
```

Comment:
```suggestion
#if defined(CONFIG_NVMEM)
	struct nvmem_cell cell;
#endif
```

should not increase size if not needed